app_check/src/ios/app_attest_provider_ios.mm

```diff
@@ -12,10 +12,69 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "app_check/src/ios/app_attest_provider_ios.h"
+
 #include "firebase/app_check/app_attest_provider.h"
 
+#import "FIRAppAttestProvider.h"
+
+#include "app/src/util_ios.h"
+#include "app_check/src/ios/app_check_ios.h"
+#include "app_check/src/ios/util_ios.h"
+#include "firebase/app_check.h"
+
 namespace firebase {
 namespace app_check {
+namespace internal {
+
+class AppAttestProvider : public AppCheckProvider {
+ public:
+  AppAttestProvider(FIRAppAttestProvider* provider);
+  virtual ~AppAttestProvider();
+
+  /// Fetches an AppCheckToken and then calls the provided callback method with
+  /// the token or with an error code and error message.
+  virtual void GetToken(
+      std::function<void(AppCheckToken, int, const std::string&)> completion_callback) override;
+
+ private:
+  FIRAppAttestProvider* ios_provider_;
+};
+
+AppAttestProvider::AppAttestProvider(FIRAppAttestProvider* provider) : ios_provider_(provider) {}
+
+AppAttestProvider::~AppAttestProvider() {}
+
+void AppAttestProvider::GetToken(
+    std::function<void(AppCheckToken, int, const std::string&)> completion_callback) {
+  [ios_provider_
+      getTokenWithCompletion:^(FIRAppCheckToken* _Nullable token, NSError* _Nullable error) {
+        completion_callback(firebase::app_check::internal::AppCheckTokenFromFIRAppCheckToken(token),
+                            firebase::app_check::internal::AppCheckErrorFromNSError(error),
+                            util::NSStringToString(error.localizedDescription).c_str());
+      }];
+}
+
+AppAttestProviderFactoryInternal::AppAttestProviderFactoryInternal() {}
+
+AppAttestProviderFactoryInternal::~AppAttestProviderFactoryInternal() {}
```

Comment:
It'd be good to clear out the map in the destructor.  That is, delete the Providers, and then clear it.

Reply:
I was thinking that in general the thing created by a factory can outlast the factory, but I think in the case of app check it is reasonable to assume that firebase SDK is the owner for the factory and all of the providers created by it, and the factory is not expected to be destroyed while it is still being used by app check.

So I will go ahead and delete the providers and clear the map.